### PR TITLE
Add GPIO support to PCA9685 for Navigator boards

### LINKED ISSUE #21563: Navigator Relay support
## Feature request

**Is your feature request related to a problem? Please describe.**
Pixhawk flight controllers provide some relay-style output options, but the new Blue Robotics Navigator does not. This makes it difficult for some users to change from Pixhawk to Navigator.

**Describe the solution you'd like**
Add relay support to the PWM outputs of the Navigator board, such that an arbitrary pin can be configured as a relay output instead of PWM. This is possible with the PCA9685 chip used on the Navigator.

**Describe alternatives you've considered**
- Explicit GPIO pins could be used instead, but these are very limited on the Navigator (especially when compared to the 16 available PWM outputs)
- It would also be possible to allow only a subset of pins to be usable as relays, but that seems unnecessarily restrictive given it is not a limitation of the actual hardware

**Platform**
[x] All
[  ] AntennaTracker
[  ] Copter
[  ] Plane
[  ] Rover
[x] Submarine

**Additional context**
Requested and discussed in [this forum post](https://discuss.bluerobotics.com/t/relay-integration-with-the-navigator-board/12715). 

Has also been discussed internally in Blue Robotics, and is being worked on by @Williangalvani:
- [initial attempt](https://github.com/williangalvani/ardupilot/tree/relay)
   - seemingly works
   - uses PWM duty cycle with a magic number
   - "ON" state shows as 5000 in `SERVO_OUTPUT_RAW`
   - may have signal spikes at period change (would need confirmation with an oscilloscope)
   - currently setting pin 10 maps to pin 11
- [current WIP attempt](https://github.com/williangalvani/ardupilot/tree/relay_2)
   - seemingly works
   - uses PCA9685 "always ON/OFF" bits
   - "ON" state shows as 1 in `SERVO_OUTPUT_RAW`
   - is a more logically intuitive/"correct" implementation
   - ~~currently setting pin 10 maps to pin 11 - needs correction (maybe a zero-indexing issue or something?)~~ fixed

### PR DESCRIPTION
**Pull request by Williangalvani**
~This enables all RCOutput drivers to implement Relay support, and it is only enabled if `PCA9685_ENABLE_RELAYS` is set.~

This now adds PCA9685 GPIO functionality to Navigator boads only, via GPIO_Navigator.

It has a whitelist of gpio pins that can be used in the pi itself without interfering with sensors, and all others get mapped to the PCA9685 (pwm controller) chip.

It uses the "always on" bit that can be seen in the datasheet's Table 7 (page 21).

[Datasheet link for convenience](https://cdn-shop.adafruit.com/datasheets/PCA9685.pdf).

Tested on Navigator.

closes #21563

### CODE REVIEW

File: libraries/AP_Relay/AP_Relay.cpp [hunk 1]
```diff
@@ -141,6 +141,10 @@ void AP_Relay::set(const uint8_t instance, const bool value)
     if (_pin[instance] == -1) {
         return;
     }
+    if (hal.rcout->supports_relays()) {
+      hal.rcout->write_relay(_pin[instance], value);
```
Comment (rmackay9) [inline libraries/AP_Relay/AP_Relay.cpp:145]: small indenting issue..

**Review by tridge — CHANGES_REQUESTED** [review]
should be done in GPIO.cpp in HAL_Linux, with magic range of pin numbers, so it works with all GPIOs, not just relay

**Comment by ES-Alexander** [comment]
@tridge / @IamPete1 out of interest, are there likely to be usability issues / unintuitive behaviour caused by presenting an output-only set of pins as GPIO? Is there some way to flag or mark a GPIO pin as input or output only?

**Comment by IamPete1** [comment]
@ES-Alexander The IOMCU GPIO are also output only, we don't really have a great way to document this ATM. We can only put them in relevant param descriptions, which is probably only relay....

**Review by tridge — APPROVED** [review]

**Pull request merged**